stellargraph/layer/deep_graph_infomax.py

```diff
@@ -127,14 +128,22 @@ def build(self):
             GlobalAveragePooling1D()(self._node_feats)
         )
 
-        discriminator = BilinearTransform()
+        discriminator = DGIDiscriminator(name=self._unique_id)
         scores = discriminator([self._node_feats, summary])
         scores_corrupted = discriminator([node_feats_corr, summary])
 
-        x_out = tf.concat([scores, scores_corrupted], axis=2)
+        x_out = tf.stack([scores, scores_corrupted], axis=2)
 
         return x_corr + x_inp, x_out
 
+    @property
+    def _unique_id(self):
+        """
+        A unique string id for this object to be used as a keras layer name.
+        """
+        id = str(self).replace(" ", "")[1:-1]
```

Comment:
I think all of this can be replaced with just the `id` function, since it's a more complicated way of extracting that information (the only unique bit in the default `str` is the hex number after `at ...` and that is exactly `hex(id(...))`): https://docs.python.org/3/library/functions.html#id

```python
>>> o = object()
>>> hex(id(o))
'0x1033872f0'
>>> str(o)
'<object object at 0x1033872f0>'
```

Could you say why you went with this approach?

Reply:
I wasn't aware of this function, cheers!